packages/expo-notifications/src/utils/updatePushTokenAsync.ts

```diff
@@ -0,0 +1,138 @@
+import { CodedError, Platform } from '@unimodules/core';
+import * as Application from 'expo-application';
+
+import { Registration } from '../DevicePushTokenAutoRegistration.fx';
+import ServerRegistrationModule from '../ServerRegistrationModule';
+import { DevicePushToken } from '../Tokens.types';
+import generateRetries from './generateRetries';
+import makeInterruptible from './makeInterruptible';
+
+export const [
+  updatePushTokenAsync,
+  hasPushTokenBeenUpdated,
+  interruptPushTokenUpdates,
+] = makeInterruptible<[DevicePushToken]>(updatePushTokenAsyncGenerator);
+
+async function* updatePushTokenAsyncGenerator(token: DevicePushToken) {
+  // Fetch the latest registration info from the persisted storage
+  const lastRegistrationInfo = yield ServerRegistrationModule.getLastRegistrationInfoAsync?.();
```

Comment:
Can we change this approach and use our standard mechanisms for detection native method unavailability? 🤔

Reply:
Done! (?) https://github.com/expo/expo/pull/10908/commits/4c3a5ce483a47fdacd320c75a528c779a6f908b4